### ISSUE TITLE
Training is slow? Converge slowly?

### ISSUE THREAD

**Issue by EvaFlower** [zhuyu-cs/MeanFlow#16]
Hi, thanks for sharing the code.

I tried to run experiments on 4 NVIDIA RTX ADA Generation (48GB) with the following script:

`accelerate launch --num_processes=4     train.py     --exp-name "cifar_unet"     --output-dir "work_dir"     --data-dir "/data/dataset/train_sdvae_latents_lmdb"     --resolution 32     --batch-size 256     --gradient-accumulation-steps 4     --allow-tf32     --mixed-precision "bf16"     --epochs 400     --path-type "linear"     --weighting "adaptive"     --time-sampler "logit_normal"     --time-mu -2.0     --time-sigma 2.0     --ratio-r-not-equal-t 0.75     --adaptive-p 0.75     --checkpointing-steps 1000`,

which runs about 20k training iterations, but needs to take around 10 hours. I wonder if such a training time is normal and how long for your case?

I also tried https://github.com/haidog-yaqub/MeanFlow, which runs around 20 hours for 200k training iterations. I also found their code can have some meaningful image results with 5 sampling steps after 500 training iterations, but your generated images with 5 sampling steps after 20k training iterations are still quite noisy. I wonder why the code here converges slowly?

Thanks in advance!

**Comment by zhuyu-cs** [comment]
Our implementation is based on the unconditional training paradigm following the MeanFlow paper. The [repo](https://github.com/haidog-yaqub/MeanFlow) you mentioned uses a different approach that doesn't employ the UNet architecture or the unconditional training paradigm described in the paper. You might find it helpful to refer to the MeanFlow paper.

**Comment by EvaFlower** [comment]
Thanks a lot for your reply! 

I got it.  BTW, could you share your training time for CIFAR and ImageNet?

**Comment by zhuyu-cs** [comment]
Certainly. On 8x A100 40G GPUs, CIFAR with 200K iterations took 3.2 days. With the same resources, SiT-B/4-wocfg took approximately 18 hours, SiT-B/4 took about 1.7 days, SiT-B/2 took around 5.6 days, SiT-L/2 took approximately 16 days. SiT-XL/2 was trained using 2 nodes and took about 17 days.

**Comment by EvaFlower** [comment]
Thank you very much!

**Comment by deJQK** [comment]
> 18 hours

Hi @zhuyu-cs , thanks for sharing the details. When training SiT-B/4-wocfg, did you set cfg-prob to 0?
